### PR TITLE
test: added errcheck tool

### PR DESCRIPTION
**Pull request by kayrus**
Implements automatic checks for issues like this #6016.

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -214,6 +214,15 @@ function dep_pass {
 	fi
 }
 
+function errcheck_pass {
+	echo "Running errcheck tests..."
+        export PATH="$PATH:$GOPATH/bin"
+        if ! which errcheck > /dev/null ; then
+                go get github.com/kisielk/errcheck || { echo "Failed to install 'errcheck'"; exit 255; }
```
Comment (heyitsanthony) [inline test]: the `go get` should be in `.travis.yml`; the pass itself probably belongs in `fmt_pass` since that's where the other static checks live

**Comment by heyitsanthony** [comment]
oof looks like etcd is missing a lot of error checks; thanks for looking into this

File: test (hunk 1 above)
Comment (kayrus) [inline test]: Moved `go get` in `.travis.yml`, but I don't think this check should be located inside the `fmt_pass`. Using dedicated function you can run only `errcheck` or skip it.

**Comment by kayrus** [comment]
@heyitsanthony fixed

**Comment by xiang90** [comment]
@kayrus Can we fix CI?

**Comment by kayrus** [comment]
@xiang90 looks like this problem is temp. rebased the changes, will see once again.

**Comment by kayrus** [comment]
@xiang90 it should fail on exact `errcheck` but log messages are weird and don't relate on errcheck. anyway, I've disabled errcheck at the moment. You can use this in the future.

**Comment by kayrus** [comment]
@xiang90 this log definetely doesn't relate to errcheck:

```sh
# github.com/coreos/etcd/cmd/etcd/cmd/vendor/google.golang.org/grpc/transport
cmd/vendor/google.golang.org/grpc/transport/http_util.go:247: internal compiler error: schedule does not include all values
goroutine 1 [running]:
runtime/debug.Stack(0x0, 0x0, 0x0)
	/home/travis/.gimme/versions/go/src/runtime/debug/stack.go:24 +0x79
cmd/compile/internal/gc.Fatalf(0xa0c844, 0x24, 0x0, 0x0, 0x0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/subr.go:165 +0x248
cmd/compile/internal/gc.(*ssaExport).Fatalf(0xce6668, 0xc4000009f7, 0xa0c844, 0x24, 0x0, 0x0, 0x0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/ssa.go:4569 +0x74
cmd/compile/internal/ssa.(*Config).Fatalf(0xc421410000, 0x9f7, 0xa0c844, 0x24, 0x0, 0x0, 0x0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/ssa/config.go:271 +0x6e
cmd/compile/internal/ssa.(*Func).Fatalf(0xc4220eb590, 0xa0c844, 0x24, 0x0, 0x0, 0x0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/ssa/func.go:397 +0x6b
cmd/compile/internal/ssa.schedule(0xc4220eb590)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/ssa/schedule.go:257 +0x11ef
cmd/compile/internal/ssa.Compile(0xc4220eb590)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/ssa/compile.go:64 +0x323
cmd/compile/internal/gc.buildssa(0xc420f4abd0, 0x0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/ssa.go:235 +0xc5e
cmd/compile/internal/gc.compile(0xc420f4abd0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/pgen.go:405 +0x1370
cmd/compile/internal/gc.funccompile(0xc420f4abd0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/dcl.go:1287 +0x186
cmd/compile/internal/gc.Main()
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/main.go:465 +0x19ab
cmd/compile/internal/amd64.Main()
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/amd64/galign.go:93 +0x2fa
main.main()
	/home/travis/.gimme/versions/go/src/cmd/compile/main.go:33 +0x2a3
github.com/coreos/etcd/cmd/etcd/cmd/vendor/github.com/prometheus/common/model
github.com/coreos/etcd/cmd/etcd/cmd/vendor/github.com/prometheus/procfs
# github.com/coreos/etcd/cmd/etcd/cmd/vendor/github.com/prometheus/common/model
cmd/vendor/github.com/prometheus/common/model/value.go:396: invalid instruction: 00649 (/home/travis/gopath/src/github.com/coreos/etcd/cmd/etcd/cmd/vendor/github.com/prometheus/common/model/time.go:180)	IMULQ	$3600000000000, AX
cmd/vendor/github.com/prometheus/common/model/value.go:396: invalid instruction: 00761 (/home/travis/gopath/src/github.com/coreos/etcd/cmd/etcd/cmd/vendor/github.com/prometheus/common/model/time.go:178)	IMULQ	$86400000000000, AX
cmd/vendor/github.com/prometheus/common/model/value.go:396: invalid instruction: 00916 (/home/travis/gopath/src/github.com/coreos/etcd/cmd/etcd/cmd/vendor/github.com/prometheus/common/model/time.go:182)	IMULQ	$60000000000, AX
```

**Comment by dongsupark** [comment]
@kayrus FWIW I just happened to see the weird internal error with `google.golang.org/grpc`. See https://travis-ci.org/coreos/fleet/jobs/152637915. First, it fails only with go tip, not with go 1.7 or older. And until yesterday it worked just fine. So it must be a corner case caused by something between go tip and grpc. After all travis is configured to be allowed to fail with go tip...

**Pull request closed without merging**